src/io/unix/sync.rs

```diff
@@ -59,39 +59,61 @@ impl File {
         self.size.load(Ordering::SeqCst)
     }
 
-    pub(crate) async fn write_append_all(&self, buf: Bytes) -> IOResult<()> {
-        let offset = self.size.fetch_add(buf.len() as u64, Ordering::SeqCst);
-        self.write_all_at(offset, buf).await
-    }
-
-    pub(crate) async fn write_append_all_buffers(
+    pub(crate) async fn write_append_writable_data<R: Send + 'static>(
         &self,
-        first_buf: Bytes,
-        second_buf: Bytes,
-    ) -> IOResult<()> {
-        let first_len = first_buf.len() as u64;
-        let second_len = second_buf.len() as u64;
-        let len = first_len + second_len;
-        let mut offset = self.size.fetch_add(len, Ordering::SeqCst);
+        c: impl WritableDataCreator<R>,
+    ) -> Result<R> {
+        let len = c.len();
+        let size = self.size.clone();
         let file = self.no_lock_fd.clone();
         if Self::can_run_inplace(len) {
             Self::inplace_sync_call(move || {
-                file.write_all_at(&first_buf, offset)?;
-                offset = offset + first_len;
-                file.write_all_at(&second_buf, offset)?;
-                Ok(())
+                let offset = size.fetch_add(len, Ordering::SeqCst);
+                let (res, data) = c.create(offset)?;
```

Comment:
This call prevent us from returning `IoResult<>`, but as I see the current implementation never returns an error, so `create` signature can be updated to `fn create(self, offset: u64) -> (WritableData, R)`